### PR TITLE
Release 1.4-rc1

### PR DESCRIPTION
**Pull request by ca-scribner**
Bundle updates:
* remove `-edge` bundle
* use kubeflow v1.4 charms/images in all bundles
* pull all charms from charmhub for all bundles

CI updates:
* use microk8s 1.21 in integrate.yaml
* publish lite and regular bundles
* Update tests to match new bundles

### CODE REVIEW

**Review by DnPlas — APPROVED** [review]

File: bundle-kubeflow-lite.yaml [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+- [istio-pilot:istio-pilot, istio-ingressgateway:istio-pilot]
+- [kfp-api, kfp-db]
+- [kfp-api:kfp-api, kfp-persistence:kfp-api]
+- [kfp-api:kfp-api, kfp-ui:kfp-api]
+- [kfp-api:kfp-viz, kfp-viz:kfp-viz]
+- [kfp-api:object-storage, minio:object-storage]
+- [kfp-profile-controller:object-storage, minio:object-storage]
+- [kfp-ui:object-storage, minio:object-storage]
+- [kubeflow-profiles, kubeflow-dashboard]
+- [mlmd:grpc, envoy:grpc]
```
Comment (DomFleischmann) [inline bundle-kubeflow-lite.yaml:48]: When trying to deploy the lite bundle, this relation does not work as envoy is missing in the bundle, we should add it.

File: bundle-kubeflow-lite.yaml [hunk 2]
```diff
@@ -0,0 +1,48 @@
+bundle: kubernetes
+name: kubeflow-lite
+applications:
+  admission-webhook:             { charm: admission-webhook,       channel: latest/edge, scale: 1 }
+  argo-controller:               { charm: argo-controller,         channel: latest/edge, scale: 1 }
+  dex-auth:                      { charm: dex-auth,                channel: 2.28/edge, scale: 1, trust: true }
```
Comment (ca-scribner) [inline bundle-kubeflow-lite.yaml:6]: Restore accidentally deleted envoy charm
```suggestion
  dex-auth:                      { charm: dex-auth,                channel: 2.28/edge, scale: 1, trust: true }
  envoy: { charm: envoy, channel: latest/edge, scale: 1 }
```

File: bundle-kubeflow.yaml [hunk 3]
```diff
@@ -0,0 +1,59 @@
+bundle: kubernetes
+name: kubeflow
+applications:
+  admission-webhook:             { charm: admission-webhook,       channel: latest/edge, scale: 1 }
+  argo-controller:               { charm: argo-controller,         channel: latest/edge, scale: 1 }
+  argo-server:                   { charm: argo-server,             channel: latest/edge, scale: 1 }
+  dex-auth:                      { charm: dex-auth,                channel: 2.28/edge, scale: 1, trust: true }
+  envoy:                         { charm: envoy,                   channel: latest/edge, scale: 1 }
+  istio-ingressgateway:          { charm: istio-gateway,           channel: 1.5/edge, scale: 1, trust: true}
```
Comment (knkski) [inline bundle-kubeflow.yaml:9]: I don't think we're planning on having `istio-gateway` be compatible with the 1.5 track - `istio-gateway` is something introduced with 1.11

Comment (DomFleischmann) [reply bundle-kubeflow.yaml:9]: In the end we decided to just publish the istio-ingressgateway into this track to make the whole thing consistent. It is still a istio gateway charm egress is just not enabled.

**Pull request merged**